module/batch-action/src/Infrastructure/Persistence/Query/DbalBatchActionQuery.php

```diff
@@ -65,6 +67,44 @@ public function getInformation(BatchActionId $id, Language $language): BatchActi
         return $model;
     }
 
+    public function getProfileInfo(): array
+    {
+        $qb = $this->connection->createQueryBuilder();
+
+        return $qb->select('id')
+            ->addSelect('(select (case
+                                            when (select bool_and(success)
+                                                  from batch_action_entry
+                                                  where batch_action_id = ba.id) then \'success\'
+                                            else \'processing\' end) as status)')
+            ->addSelect('created_at as started_at')
+            ->addSelect('(select MAX(processed_at)
+                                 from batch_action_entry
+                                 where batch_action_id = ba.id) AS ended_at')
+            ->addSelect('type as name')
+            ->addSelect('(select count(*)
+                                from batch_action_entry
+                                where batch_action_id = ba.id) as items')
+            ->addSelect('(select count(*)
+                                from batch_action_entry
+                                where batch_action_id = ba.id
+                                  and success = true) as processed')
+            ->addSelect('(select count(*)
+                                 from batch_action_entry
+                                 where batch_action_id = ba.id
+                                   and success = true
+                                   and fail_reason is null) as succeeded')
+            ->addSelect('(select count(*)
+                                 from batch_action_entry
+                                 where batch_action_id = ba.id
+                                   and fail_reason is not null) as errors')
+            ->orderBy('started_at', 'DESC')
+            ->from(self::TABLE_BATCH_ACTION, 'ba')
+            ->setMaxResults(10)
```

Comment:
```suggestion
            ->setMaxResults(self::PROFILE_RESOULT)
```
Magick number to const